### DISCUSSION TITLE
How to use contains filter in array from foreign table?

### DISCUSSION THREAD

**Discussion by franciscolourenco** [supabase/supabase#11468] [Questions]
I want to select all user profiles which are not in the current project.
However I get the following error: `column profiles.projects does not exist`
```javascript
const { data, error } = await supabase
      .from('profiles')
      .select('id, email, projects!inner(id)')
      .not('projects','cs' , `{'${route.params.id}'}`)
```

`eq` filters work fine, but when trying `contains`/`cs` filters, it can't find the foreign relationship.

**Comment by GaryAustin1** [comment]
Projects looks like a table.   Contains works on an array column to compare to an array of values.
Can you show your eq that worked?  Was it on projects also?

**Reply by franciscolourenco** [reply]
This works, but doesn't quite produce the same outcome that `contains` would:

```javascript
const { data, error } = await supabase
      .from('profiles')
      .select('id, email, projects!inner(id)')
      .not('projects.id', 'eq' , route.params.id)
```

**Reply by GaryAustin1** [reply]
So products.id is a single value column. If you put contains/cs in the filter you need a products.arraycol for it to work on. You can't just use the tablename.

**Reply by GaryAustin1** [reply]
You may be trying to do "anti-joins" here https://supabase.com/blog/postgrest-11-prerelease. 
Scroll down to it.
I'm not sure PostgREST 11 has been rolled out to the hosted instances yet.
